Octokit.Tests.Integration/Reactive/ObservableCommitStatusClientTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Reactive.Linq;
+using System.Threading.Tasks;
+using Octokit.Reactive;
+using Xunit;
+
+namespace Octokit.Tests.Integration.Reactive
+{
+    public class ObservableCommitStatusClientTests
+    {
+        public class TheGetAllMethod
+        {
+            readonly ObservableCommitStatusClient _commitStatusClient;
+            const string owner = "octokit";
+            const string name = "octokit.net";
+            const string reference = "master";
+
+            public TheGetAllMethod()
+            {
+                var github = Helper.GetAuthenticatedClient();
+                _commitStatusClient = new ObservableCommitStatusClient(github);
+            }
+
+            [IntegrationTest]
+            public async Task ReturnCommitStatus()
+            {
+                var commitStatus = await _commitStatusClient.GetAll(owner, name, reference).ToList();
+
+                Assert.NotEmpty(commitStatus);
+            }
+
+            [IntegrationTest]
+            public async Task ReturnsCorrectCountOfCommitStatusWithoutStart()
+            {
+                var options = new ApiOptions
+                {
+                    PageSize = 5,
+                    PageCount = 1
+                };
+
+                var commitStatus = await _commitStatusClient.GetAll(owner, name ,reference , options).ToList();
+
+                Assert.Equal(5, commitStatus.Count);
+            }
+
+            [IntegrationTest]
+            public async Task ReturnsCorrectCountOfCommitStatusWithStart()
+            {
+                var options = new ApiOptions
+                {
+                    PageSize = 5,
+                    PageCount = 1,
+                    StartPage = 1
+                };
+
+                var commitStatus = await _commitStatusClient.GetAll(owner, name, reference, options).ToList();
+
+                Assert.Equal(5, commitStatus.Count);
+            }
+
+            [IntegrationTest]
+            public async Task ReturnsDistinctResultsBasedOnStartPage()
+            {
+                var startOptions = new ApiOptions
+                {
+                    PageSize = 5,
+                    PageCount = 1
+                };
+
+                var firstPage = await _commitStatusClient.GetAll(owner, name, reference, startOptions).ToList();
+
+                var skipStartOptions = new ApiOptions
+                {
+                    PageSize = 5,
+                    PageCount = 1,
+                    StartPage = 2
+                };
+
+                var secondPage = await _commitStatusClient.GetAll(owner, name, reference,skipStartOptions).ToList();             
+
+                Assert.NotEqual(firstPage[0].Id, secondPage[0].Id);
```

Comment:
With this one here, we just want to ensure we're getting different results. Working with a page size of two should get this passing as well, just drop the index checks that are outside this range.